### PR TITLE
SAT: basic read on full catalog when `test_strictness_level == high`

### LINKED ISSUE #18306: SAT[strict_mode]: `basic_read` should use the discovered catalog at least once
## Tell us about the problem you're trying to solve
User defined catalog can be used on the `basic_read` test.
Some streams can be omitted from these custom catalog, leading to partial coverage of `basic_read`.
I would like to ensure `basic_read` test is run at least once with the discovered

Current behavior:
In `acceptance-test-catalog.yml` when `configured_catalog_path` is not set under `basic_read`, `basic_read` will use the discovered catalog: 
```yaml
tests:
  basic_read:
    - config_path: "integration_tests/config.json"
```
But users can defined a custom configured catalog that  can omit some streams
```yaml
tests:
  basic_read:
    - config_path: "integration_tests/config.json"
      configured_catalog_path: "integration_tests/configured_catalog.json"
```

## Suggested solution

**Solution A**:  <-- Preference 
1. When `test_mode == strict` run all basic_read test twice: once with the if the custom `configured_catalog` and once with the discovered_catalog. Modification to `configured_catalog` or `discovered_catalog` fixtures could help achieve this.

**Solution B:**
1. Expose a new fixture that parses all the basic_read test configurations and checks if `configured_catalog_path is None` for one of these tests.
3. `test_read`uses this fixture, if we are in strict mode and the fixture returns `false` the test fails with the reason "Strict mode: test_read should run at least once the discovered catalog"

### PR DESCRIPTION
**Pull request by alafanechere**
## What
Closes https://github.com/airbytehq/airbyte/issues/18306
When `test_strictness_level` is `high` we want to discard any custom configured catalog passed and use the discovered catalog instead, from which all the declared `empty_streams` are removed.

## How
* Refacto: change `test_read` signature by splitting `inputs`'s attributes into granular fixtures (on which strictness level is enforced)
* Work on the `configured_catalog_fixture` to not use any custom configured catalog when the strictness is high. We build the configured catalog from the discovered catalog from which we remove the declared `empty_streams`.
* Make the use of custom catalog for testing safer: all stream declared in the configured catalog must exists in the discovered catalog

## Recommended reading order
1. https://github.com/airbytehq/airbyte/blob/5acd80df090be68324a0cf22c32e038b57689673/airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/conftest.py#L78
2. https://github.com/airbytehq/airbyte/blob/5acd80df090be68324a0cf22c32e038b57689673/airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py#L486
3. https://github.com/airbytehq/airbyte/blob/5acd80df090be68324a0cf22c32e038b57689673/airbyte-integrations/bases/source-acceptance-test/unit_tests/test_global_fixtures.py#L170
4. https://github.com/airbytehq/airbyte/blob/5acd80df090be68324a0cf22c32e038b57689673/airbyte-integrations/bases/source-acceptance-test/unit_tests/test_core.py#L262

## 🚨 User Impact 🚨
This should not impact existing configuration as they are not yet migrated to `high` test strictness level.

### CODE REVIEW

**Review by sherifnada — CHANGES_REQUESTED** [review]
Could we be transparent to the developer about this? i.e: they shouldn't be able to pass in a configured_catalog if strictness is high. Otherwise it will lead to confusing test results. Otherwise the rest of the PR looks good!

**Comment by alafanechere** [comment]
I realized the changes I was about to make on the `configured_catalog_fixture` would also affects other tests using it (full_refresh, incremental). I don't want these tests to be impacted by the test strictness level yet. I decided to implement move my changes to a new `configured_catalog_fixture` declared in the scope of `TestBasicRead` to prevent this.
This involved moving other helper functions out from `conftest` to `utils.commons`.

**Comment by alafanechere** [comment]
@sherifnada I changed to fail the test if a custom configured catalog is passed in the `basic_read` config while we are on a `high` test strictness level.

**Comment by alafanechere** [comment]
☝️ the test above should fail because a `configured_catalog_path` is defined in the `basic_read` configuration.

**Comment by alafanechere** [comment]
☝️ I removed `configured_catalog_path` for the `basic_read` configuration.
The test should pass

File: airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py [hunk 1]
```diff
@@ -483,6 +488,36 @@ def should_validate_data_points_fixture(self, inputs: BasicReadTestConfig) -> Bo
...
+            configured_catalog_path (Optional[str]): Path to a JSON file containing a custom configured catalog.
+            discovered_catalog (MutableMapping[str, AirbyteStream]): The discovered catalog.
+            empty_streams (Set[EmptyStreamConfiguration]): The empty streams declared in the test configuration.
+
+        Returns:
+            ConfiguredAirbyteCatalog: the configured Airbyte catalog.
+        """
+        if test_strictness_level is Config.TestStrictnessLevel.high or not configured_catalog_path:
+            if configured_catalog_path:
+                pytest.fail("High strictness level error: you can't set a custom configured catalog on the basic read test.")
```
Comment (sherifnada) [inline airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py:516]:
```suggestion
                pytest.fail("High strictness level error: you can't set a custom configured catalog on the basic read test when strictness level is high.")
```

**Review by sherifnada — APPROVED** [review]

**Pull request merged**